### PR TITLE
Add Spring Boot / Hazelcast sample project

### PR DESCRIPTION
**Pull request by elliedori**
This PR adds a sample project that uses Spring Boot & Hazelcast. This uses the Spring Boot / Redis example as its base and pulls configuration from the Spring / Hazelcast project. 

I've confirmed the session is being set correctly within the Hazelcast instance via running locally, logging in, and then using the Hazelcast API to inspect and then delete the session. Upon deletion, the page resets to the login page as expected.

### CODE REVIEW

File: spring-session-samples/spring-session-sample-boot-hazelcast/spring-session-sample-boot-hazelcast.gradle [hunk 1]
```diff
@@ -0,0 +1,22 @@
+apply plugin: 'io.spring.convention.spring-sample-boot'
+
+dependencies {
+    compile project(':spring-session-hazelcast')
```
Comment (eleftherias) [inline spring-session-samples/spring-session-sample-boot-hazelcast/spring-session-sample-boot-hazelcast.gradle:4]: There is some inconsistent indentation here.

File: spring-session-samples/spring-session-sample-boot-hazelcast/src/integration-test/java/sample/BootTests.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.htmlunit.webdriver.MockMvcHtmlUnitDriverBuilder;
+
+/**
+ * @author Eddú Meléndez
```
Comment (eleftherias) [inline spring-session-samples/spring-session-sample-boot-hazelcast/src/integration-test/java/sample/BootTests.java:39]: Please update the author tag with your name, or if you prefer, you can remove the tag completely.

File: spring-session-samples/spring-session-sample-boot-hazelcast/src/integration-test/java/sample/BootTests.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2014-2019 the original author or authors.
```
Comment (eleftherias) [inline spring-session-samples/spring-session-sample-boot-hazelcast/src/integration-test/java/sample/BootTests.java:2]: Please update the copyright dates to go up to 2020.

File: spring-session-samples/spring-session-sample-boot-hazelcast/spring-session-sample-boot-hazelcast.gradle (hunk 1 above)
Comment (elliedori) [reply spring-session-samples/spring-session-sample-boot-hazelcast/spring-session-sample-boot-hazelcast.gradle:4]: This is so strange – when I look at this in IntelliJ, I don't see the weird spacing. Not sure why my IDE isn't reflecting this. Thanks for catching it

**Comment by elliedori** [comment]
@eleftherias Feedback addressed, this is ready for re-review

**Comment by eleftherias** [comment]
Merged via 46d1205ff9a74e180b3d2f9efe6f1fb8a09ad100

**Pull request closed without merging**